daemon/cluster/cluster.go

```diff
@@ -1021,6 +1022,9 @@ func (c *Cluster) GetServices(options apitypes.ServiceListOptions) ([]types.Serv
 // TODO(nishanttotla): After the packages converge, the function must
 // convert distreference.Named -> distreference.Canonical, and the logic simplified.
 func (c *Cluster) imageWithDigestString(ctx context.Context, image string, authConfig *apitypes.AuthConfig) (string, error) {
+	if _, err := digest.ParseDigest(image); err == nil {
```

Comment:
This will validate something of the form:

```
sha256:e02e811dd08fd49e7f6032625495118e63f597eb150403d02e3238af1df240ba
```

`distreference.ParseNamed` followed by the `Canonical` check will validate something of the form:

```
busybox@sha256:29f5d56d12684887bdfa50dcd29fc31eea4aaf4ad3bec43daf19026a7ce69912
```